src/utils/idl_manager.py

```
@@ -0,0 +1,348 @@
+            instruction_names = parser.get_instruction_names()
+            event_names = parser.get_event_names()
+            
+            return {
+                "platform": platform.value,
+                "has_idl_support": True,
+                "instructions": instruction_names,
+                "events": event_names,
+                "instruction_count": len(instruction_names),
+                "event_count": len(event_names),
+            }
+        except Exception as e:
```

Ruff (0.12.2)

Do not catch blind exception: `Exception`

(BLE001)